traefik.md

```diff
@@ -0,0 +1,75 @@
+1. Secret handling: Secrets are not saved inside env files, but are encrypted on disk and only available to the container they are attached to.
+2. Scalability and future failover safe deployments: By using Docker swarm you can quite easily add a second proxy node and use e.g. [keepalived](https://www.keepalived.org/) as ingress.
+3. Features: Traefik comes with many options for middlewares and Plugins. If you would like, for instance, to block certain IP ranges, this could be easily done using a [plugin](https://plugins.traefik.io/plugins).
+
+## Basics
+The default user is `rocky`  
+Traefiks logs are in `/var/log/traefik`
+To have a pretty output use `hl`:
+~~~
+sudo ./hl /var/log/traefik/traefik./glog --follow
+~~~
+The machine is a ESXi VM. The University provides this fancy [dashboard](https://vcsa-rz.intra.uni-freiburg.de/). Log in with your university handle and password.
+### Docker swarm basics
+`docker service ls` shows the "services" which are similar to the ones in kubernetes.  
+`docker service rm` deletes the service and all its respective containers, in case you would like to start from clean slate.  
+`docker service logs` would not work with Traefik, because it writes directly to `/var/log/traefik`  
+
+
+## New subdomains
+To add new subdomains, add a new line to the file `files/traefik/rules/template-subdomains.yml` in the [infrastructure-playbook](https://github.com/usegalaxy-eu/infrastructure-playbook) repo. The language there is a `go` template for a `yaml` file, which might look similar to ansible at first. (In case you wonder about the syntax).  
+The line should look like the ones above, like this scheme:
+~~~
+{{template "subdomain" "<your-subdomain-name>"}}
+~~~
+Be careful, the word `subdomain` in the second colum needs to stay literaly "subdomain", only the 3rd column is changed to the new subdomain,but without any `.usegalaxy.eu`.  
+Once this is deployed, Traefik will automatically create a router for it and fetch certificates for the subdomain as well as a wildcard certificate for ITs.  
+If you did everything correctly, the new router appears on Traefik's [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/).
+
+## How to debug
+### 🚑 Galaxy not reachable
+- Go to [aws](https://signin.aws.amazon.com/) and sign in using the credentials in the vault (`aws.yml`)
+- Navigate to `route53`
+- click on `hosted zones`
+- then on `usegalaxy.eu`
+- change the `A Record` for `usegalaxy.eu` and point it to sn06's public IP address (`132.230.223.239`)
+- The record usually has a TTL of 7200s, which means, that after 2h all requests should get sn06's IP instead of Traefik's.
+- In order to bridge that time, you can install nginx on Traefik and `proxy_pass` all requests to one headnode directly.
+### usegalaxy.eu /subdomain is showing a plain `404 not found`
+Most likely something happened to the router.
+- Check the [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
```

Comment:
Typo?

```suggestion
- Check the [dashboard](https://traefik.gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
```

Reply:
yes, a result of editing too much with vim